rundeckapp/grails-app/controllers/rundeck/controllers/FrameworkController.groovy

```diff
@@ -3500,5 +3510,15 @@ class FrameworkController extends ControllerBase implements ApplicationContextAw
         }
         render(status: HttpServletResponse.SC_NO_CONTENT)
     }
+
+    @CompileStatic
+    private def nodeSourceFileExceededMaxSize(ByteArrayInputStream bais){
+        def nodesMaxSize = 65472
+        if( bais.available() > nodesMaxSize ){
+            return [result: "Nodes exceed the maximum capacity of subscribed nodes. Max: ${nodesMaxSize}, Input: ${bais.available()}"]
+        }else{
+            return null
```

Comment:
As already stated, is very counterintuitive that this method returns null if the validation is ok, and an object if the validation fails. In this case would be better to just always return an object with a `passed` flag or something alike.
Example:
`if(exeeds) return [result: false, message: "this file is wrong"]
else return [result: true, message: null]`

Reply:
WIP